explorer/templates.go

```diff
@@ -151,6 +151,56 @@ func float64Formatting(v float64, numPlaces int, useCommas bool, boldNumPlaces .
 	return []string{integer, dec[:places], dec[places:], trailingZeros}
 }
 
+func amountAsDecimalPartsTrimmed(v, numPlaces int64, useCommas bool) []string {
+
+	// Filter numPlaces to only allow up to 8 decimal places trimming (eg. 1.12345678)
+	if numPlaces > 8 {
+		numPlaces = 8
+	}
+
+	// Separate values passed in into int.dec parts.
+	intpart := v / 1e8
+	decpart := v % 1e8
+
+	// Format left side.
+	left := strconv.FormatInt(intpart, 10)
+	rightWithTail := fmt.Sprintf("%08d", decpart)
+
+	// Reduce precision according to numPlaces.
+	if len(rightWithTail) > int(numPlaces) {
+		rightWithTail = rightWithTail[0:numPlaces]
+	}
+
+	// Separate trailing zeros.
+	right := strings.TrimRight(rightWithTail, "0")
+	tail := strings.TrimPrefix(rightWithTail, right)
+
+	// For shorter values sent from system
+	if intpart == 0 {
+
+		// Do not leave value as 1. if it is case with nothing trailing
+		if (len(right)) == 0 {
+			tail = "00"
```

Comment:
I'm still wondering if you meant to put `if (len(right)) == 0 {
			tail = "00"` by itself (not in the `if intpart == 0`).  Consider `left="1"` and `right="0"`.  It will never get to this `tail = "00"` statement because of `intpart` will be 1.

Reply:
You are correct I don’t think this code gets reached anymore mainly kept it for the 0.0 case to be 0.00 but I think that is fine. Shall I remove?

Reply:
I do think it is a good idea to change "1." into "1.00", and "0." into "0.00".  It'll get reached in both cases by moving the len(right) check or eliminating the outer check.